CRM/Contribute/Form/Contribution/ThankYou.php

```diff
@@ -284,6 +283,28 @@ public function buildQuickForm() {
       $this->assign('friendURL', $url);
     }
 
+    $isPendingOutcome = TRUE;
+    try {
+      // A payment notification update could have come in at any time. Check at the last minute.
+      $contributionStatusID = civicrm_api3('Contribution', 'getvalue', array(
+        'id' => CRM_Utils_Array::value('contributionID', $params),
+        'return' => 'contribution_status_id',
+        'invoice_id' => CRM_Utils_Array::value('invoiceID', $params),
+      ));
+      if (CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $contributionStatusID) === 'Pending'
+        && !empty($params['payment_processor_id'])
```

Comment:
Yeah, currently false is kinda the default & then if there is a payment processor & it is pending then we change to pending. I just double checked & the text is fine when pay later is in use